lab3/main.py

```diff
@@ -0,0 +1,71 @@
+
+def max_equal(arr, n):
+    maxLen = Len = 1
+
+    for i in range(1, len(arr)):
+        if arr[i] == arr[i-1]:
+            Len = Len + 1
+        else:
+            maxLen = max(maxLen, Len)
+            Len = 1
+
+    maxLen = max(maxLen, Len)
+    return maxLen
+
+
+
+
+def max_ascending(arr, n):
+    maxLen = Len = 1
+
+    for i in range(1, len(arr)):
+        if arr[i] > arr[i-1]:
+            Len = Len + 1
+        else:
+            maxLen = max(maxLen, Len)
+            Len = 1
+    maxLen = max(maxLen, Len)
+    return maxLen
+
+
+
+
+if __name__ == '__main__':
+
+    menu = {}
+    menu['1'] = "Citirea unei liste de numere intregi"
+    menu['2'] = " Gasirea secventelor de lungime maxima - toate elementele egale"
+    menu['3'] = " Gasirea secventelor de lungime maxima - in ordine crescatoare"
+    menu['4'] = "Iesire"
+
+    while True:
+        for options in menu:
+            print(options, end=" ")
+            print(menu[options])
+
+        selected = int(input("Introduceti un numar:"))
+
+        if selected == 1:
```

Comment:
Regarding your implementation, a better solution with more UX would be to have a menu that is displayed after each operation.

Perhaps options 2 and 3 should be available only after an array is provided

In your case, option 4: Quit is pretty useless.